neo/SmartContract/ContractManifest.cs

```diff
@@ -0,0 +1,220 @@
+        /// A group represents a set of mutually trusted contracts. A contract will trust and allow any contract in the same group to invoke it, and the user interface will not give any warnings.
+        /// The group field can be null.
+        /// </summary>
+        public ContractManifestGroup[] Groups { get; set; }
+
+        /// <summary>
+        /// The features field describes what features are available for the contract.
+        /// </summary>
+        [JsonConverter(typeof(FeaturesConverter))]
+        public ContractPropertyState Features { get; set; }
+
+        /// <summary>
+        /// For technical details of ABI, please refer to NEP-3: NeoContract ABI. (https://github.com/neo-project/proposals/blob/master/nep-3.mediawiki)
+        /// </summary>
+        public ContractAbi Abi { get; set; }
+
+        /// <summary>
+        /// The permissions field is an array containing a set of Permission objects. It describes which contracts may be invoked and which methods are called.
+        /// </summary>
+        [JsonConverter(typeof(WillCardJsonConverter<ContractPermission>))]
+        public WildCardContainer<ContractPermission> Permissions { get; set; }
+
+        /// <summary>
+        /// The trusts field is an array containing a set of contract hashes or group public keys. It can also be assigned with a wildcard *. If it is a wildcard *, then it means that it trusts any contract.
+        /// If a contract is trusted, the user interface will not give any warnings when called by the contract.
+        /// </summary>
+        [JsonConverter(typeof(WillCardJsonConverter<UInt160>))]
+        public WildCardContainer<UInt160> Trusts { get; set; }
+
+        /// <summary>
+        /// The safemethods field is an array containing a set of method names. It can also be assigned with a wildcard *. If it is a wildcard *, then it means that all methods of the contract are safe.
+        /// If a method is marked as safe, the user interface will not give any warnings when it is called by any other contract.
+        /// </summary>
+        [JsonConverter(typeof(WillCardJsonConverter<string>))]
+        public WildCardContainer<string> SafeMethods { get; set; }
+
+        /// <summary>
+        /// Serialized size
+        /// </summary>
+        [JsonIgnore]
+        public int Size => ToJson().GetVarSize();
+
+        /// <summary>
+        /// Create Default Contract manifest
+        /// </summary>
+        /// <param name="hash">Hash</param>
+        /// <returns>Return default manifest for this contract</returns>
+        public static ContractManifest CreateDefault(UInt160 hash)
+        {
+            return new ContractManifest()
+            {
+                Hash = hash,
+                Permissions = WildCardContainer<ContractPermission>.CreateWildcard(),
+                Abi = new ContractAbi()
+                {
+                    Hash = hash,
+                    EntryPoint = new ContractMethodDescription()
```

Comment:
`EntryPoint = ContractMethodDescription.DefaultEntryPoint`

Reply:
And the hash?

Reply:
What do you mean?

Reply:
I was distracted 😆 , done!